astropy/modeling/physical_models.py

```diff
@@ -0,0 +1,195 @@
+
+
+class BlackBody(Fittable1DModel):
+    """
+    One dimensional blackbody model.
+
+    Parameters
+    ----------
+    temperature : :class:`~astropy.units.Quantity`
+        Blackbody temperature.
+
+    scale : :class:`~astropy.units.Quantity`
+        Scale factor
+
+    Notes
+    -----
+
+    Model formula:
+
+        .. math:: B_{\\nu}(T) = A \\frac{2 h \\nu^{3} / c^{2}}{exp(h \\nu / k T) - 1}
+
+    Examples
+    --------
+    >>> from astropy.modeling import models
+    >>> from astropy import units as u
+    >>> bb = models.BlackBody(temperature=5000*u.K)
+    >>> bb(6000 * u.AA)  # doctest: +FLOAT_CMP
+    <Quantity 1.53254685e-05 erg / (cm2 Hz s sr)>
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        from astropy.modeling.models import BlackBody
+        from astropy import units as u
+        from astropy.visualization import quantity_support
+
+        bb = BlackBody(temperature=5778*u.K)
+        wav = np.arange(1000, 110000) * u.AA
+        flux = bb(wav)
+
+        with quantity_support():
+            plt.figure()
+            plt.semilogx(wav, flux)
+            plt.axvline(bb.nu_max.to(u.AA, equivalencies=u.spectral()).value, ls='--')
+            plt.show()
+    """
+
+    # We parametrize this model with a temperature and a scale.
+    temperature = Parameter(default=5000.0, min=0, unit=u.K)
+    scale = Parameter(default=1.0, min=0)
+
+    # We allow values without units to be passed when evaluating the model, and
+    # in this case the input x values are assumed to be frequencies in Hz.
+    _input_units_allow_dimensionless = True
+
+    # We enable the spectral equivalency by default for the spectral axis
+    input_units_equivalencies = {"x": u.spectral()}
+
+    def evaluate(self, x, temperature, scale):
+        """Evaluate the model.
+
+        Parameters
+        ----------
+        x : float, `~numpy.ndarray`, or `~astropy.units.Quantity`
+            Frequency at which to compute the blackbody. If no units are given,
+            this defaults to Hz.
+
+        temperature : float, `~numpy.ndarray`, or `~astropy.units.Quantity`
+            Temperature of the blackbody. If no units are given, this defaults
+            to Kelvin.
+
+        scale : float, `~numpy.ndarray`, or `~astropy.units.Quantity`
+            Desired scale for the blackbody.
+
+        Returns
+        -------
+        y : number or ndarray
+            Blackbody spectrum. The units are determined from the units of
+            ``scale``.
+
+        .. note::
+
+            Use `numpy.errstate` to suppress Numpy warnings, if desired.
+
+        .. warning::
+
+            Output values might contain ``nan`` and ``inf``.
+
+        Raises
+        ------
+        ValueError
+            Invalid temperature.
+
+        ZeroDivisionError
+            Wavelength is zero (when converting to frequency).
+        """
+        if not isinstance(temperature, u.Quantity):
```

Comment:
Why not attach this to `input_units`?

Reply:
Need to know if temperature is passed with units or not in the code so that the results can be passed w/o units if temp pasted w/o units (supports the fitting).